File: 백준/Silver/3986. 좋은 단어/좋은 단어.java


import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int ret = 0;


        
        for (int i = 0; i < n; i++) {
            String s = br.readLine();
            char[] arr = s.toCharArray();
            Deque<Character> stack = new ArrayDeque<>();

            for (int j = 0; j < s.length(); j++) {
                if (stack.isEmpty()) {
                    stack.push(arr[j]);
                }
                else if (stack.peek() == arr[j]) {
                    stack.pop();
                } else {
                    stack.push(arr[j]);
                }
            }

            if (stack.isEmpty()) {
                ret++;
            }
            
        }
        System.out.println(ret);
    }
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
n = int(input())

result = 0
for _ in range(n):
    stack = []
    s = input()
    for c in s:
        if not stack:
            stack.append(c)
        else:
            if stack[-1] == c:
                stack.pop()
            else:
                stack.append(c)
    
    if not stack:
        result += 1
    
print(result)
